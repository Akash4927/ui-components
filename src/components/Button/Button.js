import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

/**
 * A button that will run a callback on click
 * ```javascript
 *  <Button onClick={alert} text="Submit" />
 * ```
 */

class Button extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(ev) {
    if (this.props.onClick) {
      this.props.onClick(ev, this.props.text);
    }
  }

  render() {
    const {
      children,
      className,
      text, disabled,
      style,
      primary,
      selected,
      danger,
      type,
    } = this.props;

    const cl = classnames('weave-button', { primary, selected, danger });
    return (
      <button
        style={style}
        disabled={disabled}
        onClick={this.handleClick}
        className={className || cl}
        type={type}
      >
        {children || text}
      </button>
    );
  }
}

Button.propTypes = {
  /**
 * Callback that will be run when the button is clicked.
 */
  onClick: PropTypes.func,
  /**
 * Text that will be used as the button label.
 * If this props is provided, it will be passed back to the `onClick` handler
 */
  text: PropTypes.string,
  /**
 * Disable the button.
 */
  disabled: PropTypes.bool,
  /**
   * Render the button in $turquoise (useful for CTAs)
   */
  primary: PropTypes.bool,
  /**
   * Add styling to show the button as selected
   */
  selected: PropTypes.bool,
  /**
   * Turn the button red to indicate something bad might happen
   */
  danger: PropTypes.bool,
  /**
   * The type of button, as it relates to <form> components
   */
  type: PropTypes.string,
};

Button.defaultProps = {
  text: 'Submit'
};

export default Button;
