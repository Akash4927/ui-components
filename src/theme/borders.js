import { kebabCase, forEach } from 'lodash';

export const borderRadius = {
  none: '0',
  soft: '2px',
  circle: '50%',
};

// Collects all theme border radii as SCSS vars.
export function themeBorderRadiiAsScss() {
  const themeBorderRadii = [];

  forEach(borderRadius, (value, name) => {
    themeBorderRadii.push(`$border-radius-${kebabCase(name)}: ${value}`);
  });

  return themeBorderRadii;
}
