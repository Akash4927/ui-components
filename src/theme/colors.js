import { isString, kebabCase, forEach } from 'lodash';

export const colors = {
  // uncategorized colors
  aliceBlue: '#f0f8ff',
  amethystSmoke: '#9292b7',
  athens: '#e2e2ec',
  athensGray: '#eeeef4',
  black: '#1a1a1a',
  blueHaze: '#c0c0d5',
  comet: '#5b5b88',
  cometDark: '#5a5a7b',
  doveGray: '#737373',
  eastBay: '#4a5d87',
  gallery: '#f4f4f4',
  gunpowder: '#3c3c5b',
  kimberly: '#7d7da8',
  kimberlyDark: '#656597',
  logan: '#b1b1cb',
  mischka: '#dfdfea',
  mulledWine: '#46466a',
  mystic: '#dae4ea',
  silver: '#ccc',
  turquoise: '#037aa9',
  white: '#fff',
  whiteLilac: '#f0f0fa',
  whiteSmoke: '#fafafc',
  promQL: {
    /**
     * GHColors theme by Avi Aryan (http://aviaryan.in)
     * Inspired by Github syntax coloring
     */
    comment: '#bbbbbb',
    string: '#e3116c',
    punctuation: '#393a34',
    entity: '#36acaa',
    metricName: '#2aa198',
    attrName: '#00a4db',
    function: '#dc322f',
    deleted: '#9a050f',
    tag: '#00009f',
    // Dropdown colors,
    salmon: '#ff7c7c',
  },
  // use these for new stuff!
  primary: {
    charcoal: '#32324b',
    lavender: '#8383ac',
  },
  accent: {
    blue: '#00d2ff',
    orange: '#ff4b19',
  },
  status: {
    // Use these when colorizing elements that should indicate a state or outcome, ie Alerts.
    success: '#38bd93', // green
    error: '#c7254e', // red
    warning: '#d4ab27', // amber
    info: '#049cd7', // blue
    disabled: '#9a9a9a',
  },
  // Used by PrometheusGraph component
  graphThemes: {
    blue: [
      '#c7e9b4',
      '#7ecdbb',
      '#1eb5eb',
      '#1d91bf',
      '#235fa9',
      '#253393',
      '#084181',
    ],
    purple: [
      '#c1d4e7',
      '#9fbddb',
      '#8d95c6',
      '#8282ab',
      '#89429e',
      '#800f7a',
      '#0b0533',
    ],
    mixed: [
      '#c7e9b4',
      '#c1d4e7',
      '#7ecdbb',
      '#9fbddb',
      '#1eb5eb',
      '#8d95c6',
      '#1d91bf',
      '#8282ab',
      '#235fa9',
      '#89429e',
      '#253393',
      '#800f7a',
      '#084181',
      '#0b0533',
    ],
  },
  // Third-party specific colors - not to be used in the theme!
  thirdParty: {
    // Google single-click login
    cornflowerBlue: '#4285f4',
    azure: '#3769bb',
  },
};

// Flattens and collects all theme colors as SCSS vars
export function themeColorsAsScss() {
  const ignoreKeys = ['graphThemes'];
  const themeColors = [];

  forEach(colors, (value, name) => {
    const colorPrefix = `$color-${kebabCase(name)}`;
    if (ignoreKeys.includes(name)) return;

    if (isString(value)) {
      themeColors.push(`${colorPrefix}: ${value}`);
    } else {
      forEach(value, (innerValue, subname) => {
        if (ignoreKeys.includes(subname)) return;
        themeColors.push(`${colorPrefix}-${kebabCase(subname)}: ${innerValue}`);
      });
    }
  });

  return themeColors;
}
