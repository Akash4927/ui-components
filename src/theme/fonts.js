import { kebabCase, forEach } from 'lodash';

export const fontFamilies = {
  regular: "'proxima-nova', Helvetica, Arial, sans-serif",
  monospace: "'Roboto Mono', monospace",
};

export const fontSizes = {
  huge: '48px',
  extraLarge: '32px',
  large: '22px',
  normal: '16px',
  small: '14px',
  tiny: '12px',
};

// Collects all theme font vars as SCSS vars.
export function themeFontsAsScss() {
  const themeFonts = [];

  forEach(fontFamilies, (value, name) => {
    themeFonts.push(`$font-family-${kebabCase(name)}: ${value}`);
  });

  forEach(fontSizes, (value, name) => {
    themeFonts.push(`$font-size-${kebabCase(name)}: ${value}`);
  });

  return themeFonts;
}
