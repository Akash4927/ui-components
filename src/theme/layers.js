import { kebabCase, forEach } from 'lodash';

export const layers = {
  front: 1,
  toolbar: 2,
  alert: 3,
  notification: 4,
  dropdown: 5,
  tooltip: 6,
  modal: 7,
};

// Collects all theme z-index layers as SCSS vars
export function themeLayersAsScss() {
  const themeLayers = [];

  forEach(layers, (value, name) => {
    themeLayers.push(`$layer-${kebabCase(name)}: ${value}`);
  });

  return themeLayers;
}
