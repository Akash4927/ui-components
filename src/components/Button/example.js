import React from 'react';

import Button from '.';

export default function ButtonExample({ clickHandler }) {
  const onClick = () => clickHandler('onClick', 'Button clicked!');
  return (
    <ul>
      <li>
        <Button onClick={onClick} />
      </li>
      <li>
        <Button disabled onClick={onClick} />
      </li>
    </ul>
  );
}
