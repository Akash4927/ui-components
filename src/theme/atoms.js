import { darken } from 'polished';

import { colors } from './colors';

// component-specific
export const atoms = {
  Alert: {
    default: {
      color: colors.black,
      background: colors.gallery,
      borderColor: darken(0.15, colors.gallery),
    },
    info: {
      color: colors.white,
      background: colors.status.info,
      borderColor: darken(0.15, colors.status.info),
    },
    success: {
      color: colors.white,
      background: colors.status.success,
      borderColor: darken(0.15, colors.status.success),
    },
    warning: {
      color: colors.white,
      background: colors.status.warning,
      borderColor: darken(0.15, colors.status.warning),
    },
    error: {
      color: colors.white,
      background: colors.status.error,
      borderColor: darken(0.15, colors.status.error),
    },
  },
  Button: {
    default: {
      color: colors.black,
      background: colors.white,
      hoverColor: colors.purple800,
      hoverBackground: darken(0.05, colors.white),
    },
    primary: {
      color: colors.white,
      background: colors.turquoise,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.turquoise),
    },
    danger: {
      color: colors.white,
      background: colors.status.error,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.status.error),
    },
    disabled: {
      color: colors.doveGray,
      background: colors.gallery,
      hoverColor: colors.doveGray,
      hoverBackground: colors.gallery,
    },
  },
};
