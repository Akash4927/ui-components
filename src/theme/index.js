import { colors, themeColorsAsScss } from './colors';
import { fontSizes, fontFamilies, themeFontsAsScss } from './fonts';
import { borderRadius, themeBorderRadiiAsScss } from './borders';
import { layers, themeLayersAsScss } from './layers';
import { spacing, themeSpacingsAsScss } from './spacings';
import { overlayIconSize, textColor, themeMiscVarsAsScss } from './misc';
import { boxShadow } from './box-shadows';
import { atoms } from './atoms';

export function themeVarsAsScss() {
  const themeVariables = []
    .concat(themeColorsAsScss())
    .concat(themeLayersAsScss())
    .concat(themeFontsAsScss())
    .concat(themeBorderRadiiAsScss())
    .concat(themeSpacingsAsScss())
    .concat(themeMiscVarsAsScss());
  return `${themeVariables.join('; ')};`;
}

const theme = {
  // Component-specific
  atoms,

  // Box shadows
  boxShadow,

  // Borders
  borderRadius,

  // Colors
  colors,

  // Fonts
  fontFamilies,
  fontSizes,

  // z-index layers
  layers,

  // Spacings
  spacing,

  // Misc
  overlayIconSize,
  textColor,
};

export default theme;
