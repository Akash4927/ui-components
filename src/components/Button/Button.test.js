import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import theme from '../../../theme';

import Button from './Button';

describe('<Button />', () => {
  describe('snapshots', () => {
    it('renders default', () => {
      const tree = renderer.create(<Button text="Submit" theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders disabled', () => {
      const tree = renderer.create(<Button disabled text="Disabled" theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders primary', () => {
      const tree = renderer.create(<Button primary text="Primary" theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders selected', () => {
      const tree = renderer.create(<Button selected text="Selected" theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders danger', () => {
      const tree = renderer.create(<Button danger text="Danger" theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('runs the callback with the text prop', () => {
    const spy = jest.fn();
    const button = shallow(<Button onClick={spy} text="MyCustomText" />);
    button.simulate('click');
    expect(spy.mock.calls[0][1]).toBe('MyCustomText');
  });
});
