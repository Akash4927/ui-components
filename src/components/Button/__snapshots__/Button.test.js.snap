// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshots renders danger 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #ff0000;
  color: #fff;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #b30000;
  color: #fff;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  type={undefined}
>
  Danger
</button>
`;

exports[`<Button /> snapshots renders default 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #fff;
  color: #000;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #d7d7d7;
  color: #3c3c5b;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  type={undefined}
>
  Submit
</button>
`;

exports[`<Button /> snapshots renders disabled 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #d7d7d7;
  color: #aaaaaa;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #d7d7d7;
  color: #aaaaaa;
}

<button
  className="c0"
  disabled={true}
  onClick={[Function]}
  type={undefined}
>
  Disabled
</button>
`;

exports[`<Button /> snapshots renders primary 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #00bcd4;
  color: #fff;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #007888;
  color: #fff;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  type={undefined}
>
  Primary
</button>
`;

exports[`<Button /> snapshots renders selected 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0px 0px 2px 2px #8383ac;
  background: #fff;
  color: #000;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #d7d7d7;
  color: #3c3c5b;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  type={undefined}
>
  Selected
</button>
`;
